extensions/applicationinsights-cfgsync-js/src/CfgSyncPlugin.ts

```diff
@@ -0,0 +1,412 @@
+/**
+* CfgSyncPlugin.ts
+* @copyright Microsoft 2018
+*/
+
+import dynamicProto from "@microsoft/dynamicproto-js";
+import { IConfig } from "@microsoft/applicationinsights-common";
+import {
+    BaseTelemetryPlugin, IAppInsightsCore, IConfigDefaults, IConfiguration, IPlugin, IProcessTelemetryContext,
+    IProcessTelemetryUnloadContext, ITelemetryItem, ITelemetryPluginChain, ITelemetryUnloadState, createProcessTelemetryContext,
+    createUniqueNamespace, eventOff, eventOn, getGlobal, getJSON, isFetchSupported, isXhrSupported, mergeEvtNamespace, onConfigChange,
+    sendCustomEvent
+} from "@microsoft/applicationinsights-core-js";
+import { doAwaitResponse } from "@nevware21/ts-async";
+import { ITimerHandler, isFunction, isNullOrUndefined, objDeepFreeze, objExtend, scheduleTimeout } from "@nevware21/ts-utils";
+import { replaceByNonOverrideCfg } from "./CfgSyncHelperFuncs";
+import {
+    ICfgSyncConfig, ICfgSyncEvent, ICfgSyncMode, NonOverrideCfg, OnCompleteCallback, SendGetFunction
+} from "./Interfaces/ICfgSyncConfig";
+import { ICfgSyncPlugin } from "./Interfaces/ICfgSyncPlugin";
+
+const evtName = "cfgsync";
```

Comment:
nit: as this is a const lets make it all uppercase.

Reply:
Actually, we should probably prefix this with something that identifies that it's use (what prefixes do we do for other events??) perhapes just `ai_` like the cookies.